### ISSUE TITLE
highly imbalanced data

### ISSUE THREAD

**Issue by geoHeil** [automl/auto-sklearn#113]
Can auto-sklearn deal with highly imbalanced data?

What about integrating a SMOTHE resampling  one-class learning strategies?

**Comment by mfeurer** [comment]
Hi,

thanks for your interest in auto-sklearn. So far, auto-sklearn can switch on a re-weighting strategy for several classifiers like SVM or RF when they are implemented in scikit-learn. Also, you have to change the metric you're optimizing for to for example F1-score. Let us know whether this works for you.

auto-sklearn uses the scikit-learn pipeline object which has problems with highly imbalanced data, see for example [this issue](https://github.com/scikit-learn/scikit-learn/issues/4617). Therefore, we can't include SMOTE right now, but we'd be happy to include it into auto-sklearn once it's included in scikit-learn.

**Comment by mfeurer** [comment]
Hi,

thanks for your interest in auto-sklearn. So far, auto-sklearn can switch on a re-weighting strategy for several classifiers like SVM or RF when they are implemented in scikit-learn. Also, you have to change the metric you're optimizing for to for example F1-score. Let us know whether this works for you.

auto-sklearn uses the scikit-learn pipeline object which has problems with highly imbalanced data, see for example [this issue](https://github.com/scikit-learn/scikit-learn/issues/4617). Therefore, we can't include SMOTE right now, but we'd be happy to include it into auto-sklearn once it's included in scikit-learn.

**Comment by geoHeil** [comment]
Thanks. I will look into sk-learn's documentation.
